### PR TITLE
feat: #361 refactored code

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

File: mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx [hunk 1]
```diff
@@ -56,9 +56,14 @@ export const renderModalContent = (params: RenderModalContentParams) => {
...
  */
-const getName = (mentors: Map<string, UserPreview>, mentorUuid: string, ownerName: string) => {
-  const mentor = mentors.get(mentorUuid);
-  const name = mentor ? mentor.name : ownerName;
+const getName = (
+  mentors: Map<string, UserPreview>,
+  formerMentors: Map<string, UserPreview>,
+  mentorUuid: string,
+  ownerName: string) => {
+  const mentorName = mentors.get(mentorUuid) || formerMentors.get(mentorUuid);
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx:64]: what does mean  ||  in current context?

File: mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx [hunk 2]
```diff
@@ -56,9 +56,14 @@ export const renderModalContent = (params: RenderModalContentParams) => {
...
+
+  const name = mentorName ? mentorName.name : ownerName;
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx:66]: lets improve logice genName, for now it spreaded across few lines. I propose to put all users in some structure like map and use it to access appropriate name

File: mw-webapp/src/model/businessModel/Way.ts [hunk 3]
```diff
@@ -39,6 +39,13 @@ export class Way {
    */
   public mentors: Map<string, UserPreview>;
 
+  /**
+   * Former mentors of this way
+   * @key @User.uuid
+   * @value @UserPreview
+   */
+  public formerMentors: Map<string, UserPreview>;
```
Comment (tritonJS826) [inline mw-webapp/src/model/businessModel/Way.ts:47]: What will happend if I became mentor second time?

Comment (Ekaterina1994) [reply mw-webapp/src/model/businessModel/Way.ts:47]: Added logic if you back as mentor your uuid will delete from field formerMentors

File: mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx (hunk 1 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx]: Deleted

File: mw-webapp/src/logic/wayPage/MentorRequestsSection.tsx [hunk 4]
```diff
@@ -22,7 +22,9 @@ const addMentorToWay = (
 
   const mentors = way.mentors.set(newUserPreview.uuid, newUserPreview);
   const mentorRequests = way.mentorRequests.filter((item) => item !== userPreview);
-  const newWay = new Way({...way, mentors, mentorRequests});
+  way.formerMentors.delete(userPreview.uuid);
+  const formerMentors = way.formerMentors;
+  const newWay = new Way({...way, mentors, mentorRequests, formerMentors});
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/MentorRequestsSection.tsx:27]: Do you need to create new instance if you already chebnged previous one?

File: mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx [hunk 5]
```diff
@@ -53,13 +53,31 @@ export const renderModalContent = (params: RenderModalContentParams) => {
   isAccepted && params.onOk();
 };
 
+/**
+ * Get all users involved in the {@link way}
+ */
+const getUsersInWay = (way: Way) => {
+  const usersInWay: Map<string, UserPreview> = new Map([
+    ...way.mentors.entries(),
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx:61]: Where is the owner?

Comment (Ekaterina1994) [reply mw-webapp/src/logic/wayPage/reportsTable/WayColumns.tsx:61]: on 63 lines?

File: mw-webapp/src/logic/wayPage/MentorRequestsSection.tsx [hunk 6]
```diff
@@ -22,7 +22,11 @@ const addMentorToWay = (
 
   const mentors = way.mentors.set(newUserPreview.uuid, newUserPreview);
   const mentorRequests = way.mentorRequests.filter((item) => item !== userPreview);
-  const newWay = new Way({...way, mentors, mentorRequests});
+  way.formerMentors.delete(userPreview.uuid);
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/MentorRequestsSection.tsx:25]:
```suggestion
```

**Review by tritonJS826 — APPROVED** [review]

**Review by tritonJS826 — APPROVED** [review]

**Pull request merged**